neo/SmartContract/ContractManifest.cs

```diff
@@ -0,0 +1,220 @@
+                },
+                Features = ContractPropertyState.NoProperty,
+                Groups = null,
+                SafeMethods = WildCardContainer<string>.CreateWildcard(),
+                Trusts = WildCardContainer<UInt160>.CreateWildcard()
+            };
+        }
+
+        /// <summary>
+        /// Return true if is allowed
+        /// </summary>
+        /// <param name="manifest">Manifest</param>
+        /// <param name="method">Method</param>
+        /// <returns>Return true or false</returns>
+        public bool CanCall(ContractManifest manifest, string method)
+        {
+            if (Groups != null && manifest.Groups != null && Groups.Any(a => manifest.Groups.Any(b => a.PubKey.Equals(b.PubKey))))
+            {
+                // Same group
+                return true;
+            }
+
+            if (manifest.Trusts != null && !manifest.Trusts.IsWildcard && !manifest.Trusts.Contains(Hash))
+            {
+                // null == * wildcard
+                // You don't have rights in the contract
+                return false;
+            }
+
+            return Permissions == null || Permissions.IsWildcard || Permissions.Any(u => u.IsAllowed(manifest.Hash, method));
+        }
+
+        /// <summary>
+        /// Parse ContractManifest from json
+        /// </summary>
+        /// <param name="json">Json</param>
+        /// <returns>Return Contract manifest</returns>
+        public static ContractManifest Parse(string json)
+        {
+            return JsonConvert.DeserializeObject<ContractManifest>(json, _jsonSettings);
+        }
+
+        /// <summary>
+        /// To json
+        /// </summary>
+        /// <param name="indented">Indented</param>
+        /// <returns>Return json string</returns>
+        public string ToJson(bool indented = false)
+        {
+            return JsonConvert.SerializeObject(this, indented ? Formatting.Indented : Formatting.None, _jsonSettings);
+        }
+
+        /// <summary>
+        /// Clone
+        /// </summary>
+        /// <returns>Return a copy of this object</returns>
+        public ContractManifest Clone() => Parse(ToJson());
+
+        /// <summary>
+        /// String representation
+        /// </summary>
+        /// <returns>Return json string</returns>
+        public override string ToString() => ToJson();
+
+        public void Serialize(BinaryWriter writer)
+        {
+            writer.WriteVarString(ToJson());
+        }
+
+        public void Deserialize(BinaryReader reader)
+        {
+            var manifest = Parse(reader.ReadVarString(MaxLength));
+
+            Hash = manifest.Hash;
+            Groups = manifest.Groups;
+            Trusts = manifest.Trusts;
+            Permissions = manifest.Permissions;
+            SafeMethods = manifest.SafeMethods;
+            Abi = manifest.Abi;
+            Features = manifest.Features;
+        }
+
+        public bool Equals(ContractManifest other)
```

Comment:
Why do we need this mothed?

Reply:
Is used for unit tests, to ensure that is the same as expected

Reply:
In unit tests, we can use `manifest.ToJson() == ""`.

Reply:
So, you do want to remove al `Equals` methods?

Reply:
They are not useful except in unit tests.